app/controllers/concerns/searchable.rb

```diff
@@ -0,0 +1,47 @@
+module Searchable
+  extend ActiveSupport::Concern
+
+  private
+
+  def sort_params
+    params.require(:data).permit(
+      :collection_id, :sort_id, :text,
+      :institution_id,
+      :theme
+    )
+  end
+
+  def select_institution_id
+    # if the institution_id is != 0 then get it
+    # if the collection id is != 0 then get that collections institution id
+    # if the collection id == 0, then make the institution id also as 0
+    if sort_params[:institution_id].to_i > 0
+      sort_params[:institution_id].to_i
+    else
+      Collection.where(id: sort_params[:collection_id]).
+        first&.institution_id.to_i
```

Comment:
Is `to_i` necessary here? I'd expect the id to already be an Integer

Reply:
hey @MC-Squared , thanks, here `to_i` is to make sure the nil is covered.